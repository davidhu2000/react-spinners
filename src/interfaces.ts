import { SerializedStyles } from '@emotion/react';

export interface LengthObject {
  value: number;
  unit: string;
}

interface CommonProps {
  color?: string;
  loading?: boolean;
  css?: string | SerializedStyles;
}

export type LengthType = number | string;

export interface LoaderHeightWidthProps extends CommonProps {
  height?: LengthType;
  width?: LengthType;
}

export interface LoaderSizeProps extends CommonProps {
  size?: LengthType;
}

export interface LoaderSizeMarginProps extends LoaderSizeProps {
  margin?: LengthType;
}

export interface LoaderHeightWidthRadiusProps extends LoaderHeightWidthProps {
  margin?: LengthType;
  radius?: LengthType;
}
