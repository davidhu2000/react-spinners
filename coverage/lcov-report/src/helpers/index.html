
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/helpers</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/helpers</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>86/86</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.88% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>32/36</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>9/9</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>85/85</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="animation.ts"><a href="animation.ts.html">animation.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

<tr>
	<td class="file high" data-value="colors.ts"><a href="colors.ts.html">colors.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="47" class="abs high">47/47</td>
	<td data-value="95" class="pct high">95%</td>
	<td data-value="20" class="abs high">19/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="46" class="abs high">46/46</td>
	</tr>

<tr>
	<td class="file high" data-value="unitConverter.ts"><a href="unitConverter.ts.html">unitConverter.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="10" class="abs high">8/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Tue Jun 25 2024 21:55:12 GMT-0700 (Pacific Daylight Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    