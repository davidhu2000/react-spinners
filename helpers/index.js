(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['exports'], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports);
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports);
    global.index = mod.exports;
  }
})(this, function (exports) {
  'use strict';

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  var calculateRgba = exports.calculateRgba = function calculateRgba(color, opacity) {
    if (color[0] === '#') {
      color = color.slice(1);
    }

    if (color.length === 3) {
      var res = '';
      color.split('').forEach(function (c) {
        res += c;
        res += c;
      });
      color = res;
    }

    var rgbValues = color.match(/.{2}/g).map(function (hex) {
      return parseInt(hex, 16);
    }).join(', ');
    return 'rgba(' + rgbValues + ', ' + opacity + ')';
  };
});