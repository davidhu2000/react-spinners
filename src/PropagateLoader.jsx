/** @jsx jsx */
import React from 'react';
import PropTypes from 'prop-types';
import { keyframes, css, jsx } from '@emotion/core';
import onlyUpdateForKeys from 'recompose/onlyUpdateForKeys';

// 1.5 4.5 7.5
let distance = [1, 3, 5];

const propagate = [
  keyframes`
      25% {transform: translateX(-${distance[0]}rem) scale(0.75)} 
      50% {transform: translateX(-${distance[1]}rem) scale(0.6)}
      75% {transform: translateX(-${distance[2]}rem) scale(0.5)}
      95% {transform: translateX(0rem) scale(1)}
    `,
  keyframes`
      25% {transform: translateX(-${distance[0]}rem) scale(0.75)} 
      50% {transform: translateX(-${distance[1]}rem) scale(0.6)}
      75% {transform: translateX(-${distance[1]}rem) scale(0.6)}
      95% {transform: translateX(0rem) scale(1)}
    `,
  keyframes`
      25% {transform: translateX(-${distance[0]}rem) scale(0.75)}
      75% {transform: translateX(-${distance[0]}rem) scale(0.75)}
      95% {transform: translateX(0rem) scale(1)}
    `,
  keyframes`
      25% {transform: translateX(${distance[0]}rem) scale(0.75)}
      75% {transform: translateX(${distance[0]}rem) scale(0.75)}
      95% {transform: translateX(0rem) scale(1)}
    `,
  keyframes`
      25% {transform: translateX(${distance[0]}rem) scale(0.75)} 
      50% {transform: translateX(${distance[1]}rem) scale(0.6)}
      75% {transform: translateX(${distance[1]}rem) scale(0.6)}
      95% {transform: translateX(0rem) scale(1)}
    `,
  keyframes`
      25% {transform: translateX(${distance[0]}rem) scale(0.75)} 
      50% {transform: translateX(${distance[1]}rem) scale(0.6)}
      75% {transform: translateX(${distance[2]}rem) scale(0.5)}
      95% {transform: translateX(0rem) scale(1)}
    `
];

class Loader extends React.Component {
    style = i => {
      const { size, sizeUnit, color } = this.props;

      return css`{
            position: absolute;
            font-size: ${`${size / 3}${sizeUnit}`};
            width: ${`${size}${sizeUnit}`};
            height: ${`${size}${sizeUnit}`};
            background: ${color};
            border-radius: 50%;
            animation: ${propagate[i]} 1.5s  infinite;
            animation-fill-mode: forwards;
        }`;
    };


  wrapper = () => {
    const wrapper = css`{
            position: relative;
        }`;

    return this.props.css ? css`${wrapper};${this.props.css}` : wrapper;
  };

  render() {
    const { loading } = this.props;

    return loading ?
      <div css={this.wrapper()}>
        <div css={this.style(0)} />
        <div css={this.style(1)} />
        <div css={this.style(2)} />
        <div css={this.style(3)} />
        <div css={this.style(4)} />
        <div css={this.style(5)} />
      </div> : null;
  }
}

Loader.propTypes = {
  loading: PropTypes.bool,
  size: PropTypes.number,
  color: PropTypes.string,
  sizeUnit: PropTypes.string,
  css: PropTypes.string
};

Loader.defaultProps = {
  loading: true,
  size: 15,
  color: '#000000',
  sizeUnit: 'px',
  css: ''
};

const Component = onlyUpdateForKeys(['loading', 'color', 'size', 'sizeUnit', 'css'])(Loader);
Component.defaultProps = Loader.defaultProps;
export default Component;
