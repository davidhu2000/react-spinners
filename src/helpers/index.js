export const calculateRgba = (color, opacity) => {
  if (color[0] === '#') {
    color = color.slice(1);
  }

  if (color.length === 3) {
    let res = '';
    color.split('').forEach(c => {
      res += c;
      res += c;
    });
    color = res;
  }

  let rgbValues = color.match(/.{2}/g).map(hex => parseInt(hex, 16)).join(', ');
  return `rgba(${rgbValues}, ${opacity})`;
};

export * from './proptypes';
