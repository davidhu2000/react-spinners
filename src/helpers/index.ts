export * from "./proptypes";
export * from "./colors";
export * from "./unitConverter";
