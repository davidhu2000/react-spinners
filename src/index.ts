import BarLoaderComponent from "./BarLoader";
import BeatLoaderComponent from "./BeatLoader";
import BounceLoaderComponent from "./BounceLoader";
import CircleLoaderComponent from "./CircleLoader";
import ClimbingBoxLoaderComponent from "./ClimbingBoxLoader";
import ClipLoaderComponent from "./ClipLoader";
import ClockLoaderComponent from "./ClockLoader";
import DotLoaderComponent from "./DotLoader";
import FadeLoaderComponent from "./FadeLoader";
import GridLoaderComponent from "./GridLoader";
import HashLoaderComponent from "./HashLoader";
import MoonLoaderComponent from "./MoonLoader";
import PacmanLoaderComponent from "./PacmanLoader";
import PropagateLoaderComponent from "./PropagateLoader";
import PulseLoaderComponent from "./PulseLoader";
import PuffLoaderComponent from "./PuffLoader";
import RingLoaderComponent from "./RingLoader";
import RiseLoaderComponent from "./RiseLoader";
import RotateLoaderComponent from "./RotateLoader";
import ScaleLoaderComponent from "./ScaleLoader";
import SkewLoaderComponent from "./SkewLoader";
import SquareLoaderComponent from "./SquareLoader";
import SyncLoaderComponent from "./SyncLoader";

import {
  LoaderHeightWidthProps,
  LoaderSizeMarginProps,
  LoaderSizeProps,
  LoaderHeightWidthRadiusProps
} from "./interfaces";

export const BarLoader: React.ComponentClass<LoaderHeightWidthProps> = BarLoaderComponent;
export const BeatLoader: React.ComponentClass<LoaderSizeMarginProps> = BeatLoaderComponent;
export const BounceLoader: React.ComponentClass<LoaderSizeProps> = BounceLoaderComponent;
export const CircleLoader: React.ComponentClass<LoaderSizeProps> = CircleLoaderComponent;
export const ClimbingBoxLoader: React.ComponentClass<LoaderSizeProps> = ClimbingBoxLoaderComponent;
export const ClipLoader: React.ComponentClass<LoaderSizeProps> = ClipLoaderComponent;
export const ClockLoader: React.ComponentClass<LoaderSizeProps> = ClockLoaderComponent;
export const DotLoader: React.ComponentClass<LoaderSizeProps> = DotLoaderComponent;
export const FadeLoader: React.ComponentClass<LoaderHeightWidthRadiusProps> = FadeLoaderComponent;
export const GridLoader: React.ComponentClass<LoaderSizeMarginProps> = GridLoaderComponent;
export const HashLoader: React.ComponentClass<LoaderSizeProps> = HashLoaderComponent;
export const MoonLoader: React.ComponentClass<LoaderSizeProps> = MoonLoaderComponent;
export const PacmanLoader: React.ComponentClass<LoaderSizeMarginProps> = PacmanLoaderComponent;
export const PropagateLoader: React.ComponentClass<LoaderSizeProps> = PropagateLoaderComponent;
export const PulseLoader: React.ComponentClass<LoaderSizeMarginProps> = PulseLoaderComponent;
export const PuffLoader: React.ComponentClass<LoaderSizeProps> = PuffLoaderComponent;
export const RingLoader: React.ComponentClass<LoaderSizeProps> = RingLoaderComponent;
export const RiseLoader: React.ComponentClass<LoaderSizeMarginProps> = RiseLoaderComponent;
export const RotateLoader: React.ComponentClass<LoaderSizeMarginProps> = RotateLoaderComponent;
export const ScaleLoader: React.ComponentClass<LoaderHeightWidthRadiusProps> = ScaleLoaderComponent;
export const SkewLoader: React.ComponentClass<LoaderSizeProps> = SkewLoaderComponent;
export const SquareLoader: React.ComponentClass<LoaderSizeProps> = SquareLoaderComponent;
export const SyncLoader: React.ComponentClass<LoaderSizeProps> = SyncLoaderComponent;
