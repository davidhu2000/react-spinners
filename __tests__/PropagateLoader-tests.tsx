import * as React from "react";
import { mount, ReactWrapper } from "enzyme";
import { matchers } from "jest-emotion";
expect.extend(matchers);

import PropagateLoader from "../src/PropagateLoader";
import { LoaderSizeProps } from "../src/interfaces";
import { sizeDefaults } from "../src/helpers";

describe("PropagateLoader", () => {
  let loader: ReactWrapper;
  let props: LoaderSizeProps;
  let defaultSize: number = 15;
  let defaultColor: string = "#000000";

  it("should match snapshot", () => {
    loader = mount(<PropagateLoader />);
    expect(loader).toMatchSnapshot();
  });

  it("should contain default props if no props are passed", () => {
    props = loader.props();
    expect(props).toEqual(sizeDefaults(15));
  });

  it("should contain styles created using default props", () => {
    for (let i: number = 0; i < 6; i++) {
      expect(loader.find("div div").at(i)).toHaveStyleRule("height", `${defaultSize}px`);
      expect(loader.find("div div").at(i)).toHaveStyleRule("width", `${defaultSize}px`);
      expect(loader.find("div div").at(i)).toHaveStyleRule("background", defaultColor);
      expect(loader.find("div div").at(i)).toHaveStyleRule("font-size", `${defaultSize / 3}px`);
    }
  });

  it("should render null if loading prop is set as false", () => {
    loader = mount(<PropagateLoader loading={false} />);
    expect(loader.isEmptyRender()).toBe(true);
  });

  it("should render the correct color based on props", () => {
    let color: string = "#e2e2e2";
    loader = mount(<PropagateLoader color={color} />);

    for (let i: number = 0; i < 6; i++) {
      expect(loader.find("div div").at(i)).not.toHaveStyleRule("background", defaultColor);

      expect(loader.find("div div").at(i)).toHaveStyleRule("background", color);
    }
  });

  it("should render the correct size based on props", () => {
    let size: number = 21;
    loader = mount(<PropagateLoader size={size} />);

    for (let i: number = 0; i < 6; i++) {
      expect(loader.find("div div").at(i)).not.toHaveStyleRule("height", `${defaultSize}px`);
      expect(loader.find("div div").at(i)).not.toHaveStyleRule("width", `${defaultSize}px`);
      expect(loader.find("div div").at(i)).not.toHaveStyleRule("font-size", `${defaultSize / 3}px`);

      expect(loader.find("div div").at(i)).toHaveStyleRule("height", `${size}px`);
      expect(loader.find("div div").at(i)).toHaveStyleRule("width", `${size}px`);
      expect(loader.find("div div").at(i)).toHaveStyleRule("font-size", `${size / 3}px`);
    }
  });

  it("should render the css override based on props", () => {
    loader = mount(
      <PropagateLoader css={"position: absolute; width: 100px; height: 200px; color: blue;"} />
    );
    expect(loader).not.toHaveStyleRule("position", "relative");
    expect(loader).toHaveStyleRule("position", "absolute");
    expect(loader).toHaveStyleRule("width", "100px");
    expect(loader).toHaveStyleRule("height", "200px");
    expect(loader).toHaveStyleRule("color", "blue");
  });
});
