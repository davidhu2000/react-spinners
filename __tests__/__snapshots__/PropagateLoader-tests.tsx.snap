// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropagateLoader should match snapshot 1`] = `
@keyframes animation-0 {
  25% {
    -webkit-transform: translateX(-1rem) scale(0.75);
    -ms-transform: translateX(-1rem) scale(0.75);
    transform: translateX(-1rem) scale(0.75);
  }

  50% {
    -webkit-transform: translateX(-3rem) scale(0.6);
    -ms-transform: translateX(-3rem) scale(0.6);
    transform: translateX(-3rem) scale(0.6);
  }

  75% {
    -webkit-transform: translateX(-5rem) scale(0.5);
    -ms-transform: translateX(-5rem) scale(0.5);
    transform: translateX(-5rem) scale(0.5);
  }

  95% {
    -webkit-transform: translateX(0rem) scale(1);
    -ms-transform: translateX(0rem) scale(1);
    transform: translateX(0rem) scale(1);
  }
}

@keyframes animation-1 {
  25% {
    -webkit-transform: translateX(-1rem) scale(0.75);
    -ms-transform: translateX(-1rem) scale(0.75);
    transform: translateX(-1rem) scale(0.75);
  }

  50% {
    -webkit-transform: translateX(-3rem) scale(0.6);
    -ms-transform: translateX(-3rem) scale(0.6);
    transform: translateX(-3rem) scale(0.6);
  }

  75% {
    -webkit-transform: translateX(-3rem) scale(0.6);
    -ms-transform: translateX(-3rem) scale(0.6);
    transform: translateX(-3rem) scale(0.6);
  }

  95% {
    -webkit-transform: translateX(0rem) scale(1);
    -ms-transform: translateX(0rem) scale(1);
    transform: translateX(0rem) scale(1);
  }
}

@keyframes animation-2 {
  25% {
    -webkit-transform: translateX(-1rem) scale(0.75);
    -ms-transform: translateX(-1rem) scale(0.75);
    transform: translateX(-1rem) scale(0.75);
  }

  75% {
    -webkit-transform: translateX(-1rem) scale(0.75);
    -ms-transform: translateX(-1rem) scale(0.75);
    transform: translateX(-1rem) scale(0.75);
  }

  95% {
    -webkit-transform: translateX(0rem) scale(1);
    -ms-transform: translateX(0rem) scale(1);
    transform: translateX(0rem) scale(1);
  }
}

@keyframes animation-3 {
  25% {
    -webkit-transform: translateX(1rem) scale(0.75);
    -ms-transform: translateX(1rem) scale(0.75);
    transform: translateX(1rem) scale(0.75);
  }

  75% {
    -webkit-transform: translateX(1rem) scale(0.75);
    -ms-transform: translateX(1rem) scale(0.75);
    transform: translateX(1rem) scale(0.75);
  }

  95% {
    -webkit-transform: translateX(0rem) scale(1);
    -ms-transform: translateX(0rem) scale(1);
    transform: translateX(0rem) scale(1);
  }
}

@keyframes animation-4 {
  25% {
    -webkit-transform: translateX(1rem) scale(0.75);
    -ms-transform: translateX(1rem) scale(0.75);
    transform: translateX(1rem) scale(0.75);
  }

  50% {
    -webkit-transform: translateX(3rem) scale(0.6);
    -ms-transform: translateX(3rem) scale(0.6);
    transform: translateX(3rem) scale(0.6);
  }

  75% {
    -webkit-transform: translateX(3rem) scale(0.6);
    -ms-transform: translateX(3rem) scale(0.6);
    transform: translateX(3rem) scale(0.6);
  }

  95% {
    -webkit-transform: translateX(0rem) scale(1);
    -ms-transform: translateX(0rem) scale(1);
    transform: translateX(0rem) scale(1);
  }
}

@keyframes animation-5 {
  25% {
    -webkit-transform: translateX(1rem) scale(0.75);
    -ms-transform: translateX(1rem) scale(0.75);
    transform: translateX(1rem) scale(0.75);
  }

  50% {
    -webkit-transform: translateX(3rem) scale(0.6);
    -ms-transform: translateX(3rem) scale(0.6);
    transform: translateX(3rem) scale(0.6);
  }

  75% {
    -webkit-transform: translateX(5rem) scale(0.5);
    -ms-transform: translateX(5rem) scale(0.5);
    transform: translateX(5rem) scale(0.5);
  }

  95% {
    -webkit-transform: translateX(0rem) scale(1);
    -ms-transform: translateX(0rem) scale(1);
    transform: translateX(0rem) scale(1);
  }
}

.emotion-6 {
  position: relative;
}

.emotion-0 {
  position: absolute;
  font-size: 5px;
  width: 15px;
  height: 15px;
  background: #000000;
  border-radius: 50%;
  -webkit-animation: animation-0 1.5s infinite;
  animation: animation-0 1.5s infinite;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.emotion-1 {
  position: absolute;
  font-size: 5px;
  width: 15px;
  height: 15px;
  background: #000000;
  border-radius: 50%;
  -webkit-animation: animation-1 1.5s infinite;
  animation: animation-1 1.5s infinite;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.emotion-2 {
  position: absolute;
  font-size: 5px;
  width: 15px;
  height: 15px;
  background: #000000;
  border-radius: 50%;
  -webkit-animation: animation-2 1.5s infinite;
  animation: animation-2 1.5s infinite;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.emotion-3 {
  position: absolute;
  font-size: 5px;
  width: 15px;
  height: 15px;
  background: #000000;
  border-radius: 50%;
  -webkit-animation: animation-3 1.5s infinite;
  animation: animation-3 1.5s infinite;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.emotion-4 {
  position: absolute;
  font-size: 5px;
  width: 15px;
  height: 15px;
  background: #000000;
  border-radius: 50%;
  -webkit-animation: animation-4 1.5s infinite;
  animation: animation-4 1.5s infinite;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.emotion-5 {
  position: absolute;
  font-size: 5px;
  width: 15px;
  height: 15px;
  background: #000000;
  border-radius: 50%;
  -webkit-animation: animation-5 1.5s infinite;
  animation: animation-5 1.5s infinite;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

<Loader
  color="#000000"
  css={Object {}}
  loading={true}
  size={15}
  sizeUnit="px"
>
  <div
    className="emotion-6"
  >
    <div
      className="emotion-0"
    />
    <div
      className="emotion-1"
    />
    <div
      className="emotion-2"
    />
    <div
      className="emotion-3"
    />
    <div
      className="emotion-4"
    />
    <div
      className="emotion-5"
    />
  </div>
</Loader>
`;
